`TestSSHLoadAllCAs/TLS_routing_enabled` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4628180293/jobs/8186957141

#### Relevant snippet
```
WARNING: DATA RACE
Write at 0x00c001601aa0 by goroutine 89741:
  github.com/gravitational/teleport/lib/client.(*proxyClusterGuesser).hostKeyCallback()
      /__w/teleport/teleport/lib/client/api.go:3082 +0xec
  github.com/gravitational/teleport/lib/client.(*proxyClusterGuesser).hostKeyCallback-fm()
      <autogenerated>:1 +0x92
  golang.org/x/crypto/ssh.(*handshakeTransport).client()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:729 +0x27d
  golang.org/x/crypto/ssh.(*handshakeTransport).enterKeyExchange()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:623 +0x7d7
  golang.org/x/crypto/ssh.(*handshakeTransport).kexLoop()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:303 +0x124
  golang.org/x/crypto/ssh.newClientTransport.func2()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:137 +0x39

Previous write at 0x00c001601aa0 by goroutine 89736:
  github.com/gravitational/teleport/lib/client.(*proxyClusterGuesser).hostKeyCallback()
      /__w/teleport/teleport/lib/client/api.go:3082 +0xec
  github.com/gravitational/teleport/lib/client.(*proxyClusterGuesser).hostKeyCallback-fm()
      <autogenerated>:1 +0x92
  golang.org/x/crypto/ssh.(*handshakeTransport).client()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:729 +0x27d
  golang.org/x/crypto/ssh.(*handshakeTransport).enterKeyExchange()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:623 +0x7d7
  golang.org/x/crypto/ssh.(*handshakeTransport).kexLoop()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:303 +0x124
  golang.org/x/crypto/ssh.newClientTransport.func2()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:137 +0x39

Goroutine 89741 (running) created at:
  golang.org/x/crypto/ssh.newClientTransport()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:137 +0x4ae
  golang.org/x/crypto/ssh.(*connection).clientHandshake()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/client.go:105 +0x48c
  golang.org/x/crypto/ssh.NewClientConn()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/client.go:83 +0x28a
  github.com/gravitational/teleport/api/observability/tracing/ssh.NewClientConn()
      /__w/teleport/teleport/api/observability/tracing/ssh/ssh.go:160 +0xba8
  github.com/gravitational/teleport/lib/client.newClientConn.func1()
      /__w/teleport/teleport/lib/client/client.go:1762 +0x130

Goroutine 89736 (running) created at:
  golang.org/x/crypto/ssh.newClientTransport()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/handshake.go:137 +0x4ae
  golang.org/x/crypto/ssh.(*connection).clientHandshake()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/client.go:105 +0x48c
  golang.org/x/crypto/ssh.NewClientConn()
      /go/pkg/mod/github.com/gravitational/crypto@v0.6.0-1/ssh/client.go:83 +0x28a
  github.com/gravitational/teleport/api/observability/tracing/ssh.NewClientConn()
      /__w/teleport/teleport/api/observability/tracing/ssh/ssh.go:160 +0xba8
  github.com/gravitational/teleport/lib/client.newClientConn.func1()
      /__w/teleport/teleport/lib/client/client.go:1762 +0x130
==================
...
=== RUN   TestSSHLoadAllCAs/TLS_routing_enabled
=== PAUSE TestSSHLoadAllCAs/TLS_routing_enabled
=== CONT  TestSSHLoadAllCAs/TLS_routing_enabled
    testing.go:1446: race detected during execution of test
--- FAIL: TestSSHLoadAllCAs/TLS_routing_enabled (12.43s)
```

--- a/lib/client/api.go
+++ b/lib/client/api.go
@@ -3115,11 +3115,18 @@ func (g *proxyClusterGuesser) authMethod(ctx context.Context) ssh.AuthMethod {
 // no JumpHosts set, i.e. presumably falling back to the proxy specified in the
 // profile.
 func (tc *TeleportClient) WithoutJumpHosts(fn func(tcNoJump *TeleportClient) error) error {
-	storedJumpHosts := tc.JumpHosts
-	tc.JumpHosts = nil
-	err := fn(tc)
-	tc.JumpHosts = storedJumpHosts
-	return trace.Wrap(err)
+	tcNoJump := &TeleportClient{
+		Config:                   tc.Config,
+		localAgent:               tc.localAgent,
+		OnShellCreated:           tc.OnShellCreated,
+		eventsCh:                 make(chan events.EventFields, 1024),
+		lastPing:                 tc.lastPing,
+		dtAttemptLoginIgnorePing: tc.dtAttemptLoginIgnorePing,
+		dtAuthnRunCeremony:       tc.dtAuthnRunCeremony,
+	}
+	tcNoJump.JumpHosts = nil
+
+	return trace.Wrap(fn(tcNoJump))
 }
 
 // Logout removes certificate and key for the currently logged in user from

